KISS build docs

murlakatamenka opened the pull request:
Fixes #101

- makes iterations while working on `wayshot` faster, thus better for devs and contributors.
- makes `make` more explicit about what it does
- fixes a minor issue of `docs/*.7.gz` not being cleaned

murlakatamenka committed:
refactor(packaging): simplify building docs

Fixes #101

Why?

- no build.rs that is invoked on _any_ `cargo build/run`
- no extra dev dependency
- no intermediate `docs/*.out` files
- 3 lines of bash instead
- maximum gzip compression (-9/--best)
- docs will be built only when needed,
    i.e. on `make docs` or `make install`

To quickly check:

```sh
make TARGET_DIR=/tmp/wayshot MAN{1,7}_DIR=/tmp/wayshot install
```

murlakatamenka committed:
refactor(packaging): explicit make output

Shows exactly what is going on.

Also fixes a minor issue of not cleaning `./docs/*.7.gz`

Shinyzenith commented:
Hi @murlakatamenka. This needs to be rebased onto `freeze-feat-andreas` branch

Shinyzenith commented:
Make check earlier used to ensure that scdoc could compile the docs too. We use it in CI.

if that can be implemented here then we can go ahead with a merge.

murlakatamenka committed:
ci: simplify checking docs; bump checkout action

For `DEBIAN_FRONTEND` environment variable see `man 7 debconf`:

- https://manpages.debian.org/bookworm/debconf-doc/debconf.7.en.html#noninteractive

murlakatamenka commented:
> Make check earlier used to ensure that scdoc could compile the docs too. We use it in CI.
> 
> if that can be implemented here then we can go ahead with a merge.

Previously the docs were built into gzipped man pages during `cargo build`/`build.rs`:

https://github.com/waycrate/wayshot/blob/8f22e6da9005f7495d38d89f2430a2649c4499ef/.github/workflows/build.yml#L19

And also checked in a separate job:

https://github.com/waycrate/wayshot/blob/8f22e6da9005f7495d38d89f2430a2649c4499ef/.github/workflows/build.yml#L32-L43

---

Now that `build.rs` is gone, only documentation job checks if `scdoc` can process the files, this should be enough for CI. Gzipping the output can be considered infallible.

Ideally there should be a job or workflow that gets triggered only if those *.scd files change, because such changes are rare compared modifying the sources.

---

So far I'm done with the changes, feel free to review.

Shinyzenith commented:
Things look great. Thanks for the PR!

Shinyzenith commented:
I just realized something I forgot to ask. The code you've written is POSIX compliant right? I would rather keep away from bash extensions.
